csharp/ql/src/semmle/code/csharp/dataflow/LibraryTypeDataFlow.qll

```diff
@@ -2417,20 +2479,38 @@ class StringValuesFlow extends LibraryTypeDataFlow, Struct {
   }
 }
 
+private predicate recordConstructorFlow(Constructor c, int i, Property p) {
+  c = any(Record r).getAMember() and
+  exists(string name |
+    c.getParameter(i).getName() = name and
+    c.getDeclaringType().getAMember(name) = p
+  )
+}
+
+private class RecordConstructorFlowRequiredSummaryComponentStack extends RequiredSummaryComponentStack {
+  private SummaryComponent head;
+
+  RecordConstructorFlowRequiredSummaryComponentStack() {
+    exists(Property p |
+      recordConstructorFlow(_, _, p) and
+      head = SummaryComponent::property(p) and
+      this = SummaryComponentStack::singleton(SummaryComponent::return())
+    )
+  }
+
+  override predicate required(SummaryComponent c) { c = head }
+}
```

Comment:
It's a bit unfortunate that we need to write these. I'll always forget them. :-(

Reply:
At least you will not have to worry about this aspect once we switch over to using CSV-based summaries.